src/helpers.jl

```diff
@@ -310,41 +300,56 @@ function htpynewlam(active, state, A, R, S, x, y, s1, s2, λ, lamFinal)
     dz = A' * dy
 
     active = Int.(active)
+
+    # Initialize  
+    i1 = 0
+    i2 = 0
+
     # Find the largest allowable steps
-    I1 = findall(i -> state[active[i]] == +1 && dx[i] .< -eps(), 1:length(active))
-    I2 = findall(i -> state[active[i]] == -1 && dx[i] .> eps(),  1:length(active))
-    if !isempty(I1)
-        alfa[1], i = findmin(-x[I1] ./ dx[I1])
-        i1=I1[i]
-        k[1] = active[I1[i]]
-    end
-    if !isempty(I2)
-        alfa[2], i = findmin(-x[I2] ./ dx[I2])
-        i2=I2[i]
-        k[2] = active[I2[i]]
+    for i in 1:length(active)
```

Comment:
Best to avoid `length` if possible:

    for i in eachindex(active)

Look for similar opportunities elsewhere.

Reply:
done!